### DISCUSSION TITLE
[ 命名建議 ] LeetCode 1603. Design Parking System

### DISCUSSION THREAD

**Discussion by LPenny-github** [EngTW/English-for-Programmers#98] [Q&A] [answered]
原 issue:
https://github.com/EngTW/English-for-Programmers/issues/80

LeetCode:
https://leetcode.com/problems/design-parking-system/

---
二香 😆😅😉
徵求命名建議 orz 感激不盡！

```csharp
public class ParkingSystem {

    int[] parkingSpaces = new int[3];

    public ParkingSystem(int big, int medium, int small) {
        parkingSpaces =new int[]{big, medium, small};
    }
    
    public bool AddCar(int carType) {
        return parkingSpaces[carType -1]-- > 0;
    }
}
```

另外請問大大 `parkingSpacesCounts` 是可行的嗎？🤔

---

PS. 我還不清楚該用什麼樣的格式發問， @twy30 大大可以直接修改本文喔 👌

**Comment by twy30** [comment, marked as answer]
@LPenny-github 



> PS. 我還不清楚該用什麼樣的格式發問， @twy30 大大可以直接修改本文喔 👌

我覺得你的文章的格式清楚易懂，我想不出更好的格式 😊

---

```C#
    int[] parkingSpaces = new int[3];
```

> 另外請問大大 `parkingSpacesCounts` 是可行的嗎？🤔

比起 `paringSpaces` (「停車位 (複數)」), 這裡我覺得 `parkingSpace` + `Counts` (「停車位數量 (複數)」) 會更貼切。

進一步可以使用 `parkingSpace` + `Capacities` (「停車位承載量 (複數)」 )

---

這裡有個 count vs. capacity 的例子：

* count
  * > Gets the number of elements contained in the List<T>.
    * 此 `List<T>` 目前有幾個成員
  * https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.count?view=net-5.0
* capacity
  * > Gets or sets the total number of elements the internal data structure can hold without resizing.
    * 此 `List<T>` 目前容量有多大
  * https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.capacity?view=net-5.0

可以參考看看 😊

**Reply by LPenny-github** [reply]
感謝 @twy30 大大 orz

**Reply by LPenny-github** [reply]
可以麻煩 @twy30 大大再解釋 `parkingSpacesCounts` 和 `parkingSpaceCounts` 的不同之處嗎？
（離開 Q&A 一小時，就又再度分不出差別 😅）

有沒有什麼好方法可以逃避複數 orz <-- 懶人魂

**Reply by twy30** [reply]
@LPenny-github 

我們可以來看看「蘋果派」這個例子 😊

* 蘋果： apple (名詞)
* 派： pie (名詞)
* 蘋果派： apple pie (名詞 + 名詞)
* 一個蘋果派： 1 apple pie
* 兩個蘋果派： 2 apple pies

「蘋果派」中的「蘋果」是用來形容「派」；「兩個蘋果派」的 複數 會加在「派」上。

---

接下來看看「蘋果派食譜」：

* 蘋果派： apple pie (名詞 + 名詞)
* 食譜： recipe (名詞)
* 蘋果派食譜： apple pie recipe (名詞 + 名詞 + 名詞)
* 一份蘋果派食譜： 1 apple pie recipe
* 二份蘋果派食譜： 2 apple pie recipes

「蘋果派食譜」中的「蘋果派」是用來形容「食譜」；「二份蘋果派食譜」的 複數 會加在「食譜」上。

---

回頭來看 `parking`, `space`, `count`

* 停車 (的動作): parking (動名詞)
* 空間： space (名詞)
* 停車位： parking space (動名詞 + 名詞)
* 一個停車位： 1 parking space
* 兩個停車位： 2 parking spaces
* 數量： count (名詞)
* 停車位數量： parking space count (動名詞 + 名詞 + 名詞)
  * 停車位數量 是 「停車位 的 數量」，複數會加在「數量」上。
* 一個停車位數量： 1 parking space count
* 二個停車位數量： 2 parking space counts

可以參考看看 😊

**Reply by LPenny-github** [reply]
@twy30 感蝦 orz
